### PR TITLE
feat: add `DataFrame.to_arrow` to create Arrow Table from DataFrame

### PR DESCRIPTION
**Pull request by tswast**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Towards #735  🦕

### CODE REVIEW

File: samples/polars/noxfile.py [hunk 1]
```diff
@@ -0,0 +1,292 @@
+# Copyright 2019 Google LLC
```
Comment (chelsea-lin) [inline samples/polars/noxfile.py:1]: Is this file auto generated from a template? If not, can we update the copyright year to `2024`? Also check other new files too?

File: samples/polars/create_polars_df_with_to_arrow_test.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    # ...
+
+    # Run the query and download the results as an Arrow table to convert into
+    # a Polars DataFrame. Use ordered=False if your polars analysis is OK with
+    # non-deterministic ordering.
+    arrow_table = bf_df.to_arrow(ordered=False)
+    polars_df = polars.from_arrow(arrow_table)
+    # [END bigquery_dataframes_to_polars]
+
+    assert polars_df is not None
```
Comment (chelsea-lin) [inline samples/polars/create_polars_df_with_to_arrow_test.py:39]: Can we assert `polars_df` is equal to `bf_df`? Or maybe there are some diverse?

**Review by chelsea-lin — APPROVED** [review]
LGTM

File: samples/polars/noxfile.py (hunk 1 above)
Comment (tswast) [reply samples/polars/noxfile.py:1]: This is templated, so not much we can do.

File: samples/polars/create_polars_df_with_to_arrow_test.py (hunk 2 above)
Comment (tswast) [reply samples/polars/create_polars_df_with_to_arrow_test.py:39]: We could possibly compare shapes or some aggregate values. I'll give it a try.

**Pull request merged**